task_manager/task_manager.hh

```diff
@@ -0,0 +1,166 @@
+#pragma once
+
+#include "replica/database.hh"
+
+using task_id = utils::tagged_uuid<struct task_id_tag>;
+using is_abortable = bool_class <struct abortable_tag>;
+
+class task_manager {
+public:
+    enum class module_type {
+        repair = 0,
+        compaction
+    };
```

Comment:
It's better to avoid listing the module types, so if a new module is added, we don't have to change and recompile the task manager and all its users.